### DISCUSSION TITLE
Removing entities not obviously explained in the documentation

### DISCUSSION THREAD

**Discussion by sergeyampo** [mikro-orm/mikro-orm#2983] [Q&A]
There's no any information about how to remove entities from the database in the docs, only API reference can help with that.
There's an example down here where we're deleting all the extracted elements and leaving only one created entity as example: 
So it's absolutely not obvious why it doesn't work:
```
    const files = await em.find(BookFileEntity, { book });
      if (files.length > 0) {
        em.remove(files);
      }

      const newFile =
        em.create(BookFileEntity, {
          book,
          originalFilename,
          format,
          sizeBytes,
        });

      await em.persistAndFlush([...files, newFile]);
```
Or this:
```
    const files = await em.find(BookFileEntity, { book });
      if (files.length > 0) {
        em.remove(files);
      }

    files.push(
        em.create(BookFileEntity, {
          book,
          originalFilename,
          format,
          sizeBytes,
        }));

      await em.persistAndFlush(files);
```
But this works:
```
      const files = await em.find(BookFileEntity, { book });
      if (files.length > 0) {
        em.remove(files);
      }

      em.persist(
        em.create(BookFileEntity, {
          book,
          originalFilename,
          format,
          sizeBytes,
        }),
      );

      await em.flush();
```

**Comment by B4nan** [comment]
Well, all of this should work. Can you provide reproduction for this? Are you on latest version?

**Reply by sergeyampo** [reply]
> Well, all of this should work. Can you provide reproduction for this? Are you on latest version?

5.0.3 version, reproduction repo - no, that's why I'm in discussions 🙃

**Reply by B4nan** [reply]
Try using latest, ideally dev version

**Reply by sergeyampo** [reply]
It's problematic to upgrade packages since 24th of february =)
